lib/indexer.rb

```diff
@@ -0,0 +1,121 @@
+module Argo
+  class Indexer
+    SimpleLogJob = Struct.new(:log_val) do
+      def perform
+        sleep(rand 5)
+        job_logger = Logger.new(File.open("tmp/simple_job.log"))
+        job_logger.unknown "#{log_val}"
+      end
+    end
+
+
+    def reindex_object obj
+      doc = obj.to_solr
+      Dor::SearchService.solr.add(doc)
+    end
+
+    def reindex_pid pid
+      obj = Dor.load_instance pid
+      reindex_object obj
+    end
+
+    def reindex_pid_list pid_list, should_commit=false
+      pid_list.each do |pid|
+        reindex_pid pid
+      end
+      ActiveFedora.solr.conn.commit if should_commit
+    end
+
+    def get_pids_for_model_type model_type
+      Dor::SearchService.risearch "select $object from <#ri> where $object <fedora-model:hasModel> #{model_type}"
+    end
+
+
+
+    def reindex_all args
+      index_log = Logger.new(File.join(Rails.root, 'log', 'reindex.log'))
+      index_log.formatter = Logger::Formatter.new
+      index_log.level = ENV['LOG_LEVEL'] ? Logger::SEV_LABEL.index(ENV['LOG_LEVEL']) : Logger::INFO
+      $stdout.sync = true
+      start_time = Time.now
+
+
+      # TODO: break out pid list creation
+      $stdout.puts 'Discovering PIDs...'
+      index_log.info 'Discovering PIDs...'
+      dor_pids = []
+      solr_pids = []
+      if args[:query] != ':ALL:'
+        q = (args[:query].nil? || args[:query] == ':MISSING:') ? '*:*' : args[:query]
+        puts q
+        start = 0
+        resp = Dor::SearchService.query(q, :sort => 'id asc', :rows => 1000, :start => start, :fl => ['id'])
+        while resp.docs.length > 0
+          solr_pids += resp.docs.collect { |doc| doc['id'] }
+          start += 1000
+          $stdout.print '.'
+          resp = Dor::SearchService.query(q, :sort => 'id asc', :rows => 1000, :start => start, :fl => ['id'])
+        end
+        $stdout.puts
+        msg = "Found #{solr_pids.length} PIDs in solr."
+        $stdout.puts msg
+        index_log.info msg
+      end
+      if args[:query] =~ /:(ALL|MISSING):/
+        dor_pids = []
+        Dor::SearchService.iterate_over_pids(:in_groups_of => 1000, :mode => :group) do |chunk|
+          dor_pids += chunk
+          $stderr.print '.'
+        end
+        $stdout.puts
+        msg = "Found #{dor_pids.length} PIDs in DOR."
+        $stdout.puts msg
+        index_log.info msg
+      end
+      if dor_pids.present?
+        pids = dor_pids - solr_pids
+      else
+        pids = solr_pids
+      end
+      pids.delete_if { |pid| pid !~ /druid:/ }
+      time = Time.now - start_time
```

Comment:
Do not use `Time.now` without zone. Use one of `Time.zone.now`, `Time.now.current`, `Time.now.in_time_zone`, `Time.now.utc`, `Time.now.getlocal`, `Time.now.iso8601`, `Time.now.jisx0301`, `Time.now.rfc3339`, `Time.now.to_i`, `Time.now.to_f` instead.